cpu/cc26x0/periph/rfc.c

```diff
@@ -72,13 +58,54 @@ void rfc_irq_disable(void)
     NVIC_EnableIRQ(RF_HW_IRQN);
 }
 
+uint32_t rfc_send_cmd(uint32_t ropAddr)
+{
+    RFC_DBELL->CMDR = ropAddr;
+  /* wait for cmd ack (rop cmd was submitted successfully) */
+  while(RFC_DBELL->RFACKIFG<<31);
+  while(!RFC_DBELL->CMDSTA);
+  if (RFC_DBELL->CMDSTA == 1)
+  {
+    printf("%s: radio operation command acknowledged. \n",__FUNCTION__);
+  }
+  else
+  {
+    printf("%s: acknowledgement of radio operation command failed (0x%" PRIu32 ") \n",__FUNCTION__,RFC_DBELL->CMDSTA);
+  }
+
+  return RFC_DBELL->CMDSTA;
+}
+
+uint16_t rfc_wait_cmd_done(void* ropCmd)
+{
+  radio_op_command_t *command = (radio_op_command_t *)ropCmd;
+  uint32_t timeout_cnt = 0;
+  /* wait for cmd execution. condition on rop status doesn't work by itself (too fast?). */
+  do{
+    if(++timeout_cnt > 500000)
+    {
+      printf("%s: execution error. ROP id: 0x%" PRIx16 " status: 0x%" PRIx16 " \n",__FUNCTION__,command->commandNo,command->status);
+      return R_OP_STATUS_DONE_TIMEOUT;
+    }
+  } while(command->status < R_OP_STATUS_SKIPPED);
+  if (command->status == 0x0400) {
```

Comment:
Regarding the conditional branching, the busy waiting doesn't work without it... I've tried to use `while(command->status < R_OP_STATUS_SKIPPED)` but the program would get stuck in the loop unless I gave it something to do... That is why I added my comment. Any idea why it is happening?
Since I didn't know what to do, I took a look at Contiki and that's where I found the `timeout` approach.